### PR TITLE
newsletter module

### LINKED ISSUE #2558: Newsletter feature not working at all
Test
- profile
- indiv components
- api

### PR DESCRIPTION
**Pull request by mdesmet**
### Related issues

closes #2558

### Short description and why it's useful

Newsletter module supporting subscribe/unsubscribe (can be integrated with mailchimp module (in develop branch of vue-storefront-api) but as long as API is respected with any email provider.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

**Comment by pkarw** [comment]
Cool stuff, really like this approach. Thanks! Could you please check the failing checks and update changelog?

**Comment by mdesmet** [comment]
It seems this issue is also on the divante develop branch and not related to my commit. 

It seems to be related to using following code parts:

`once('__VUE_EXTEND_VUEX__', () => {
  Vue.use(Vuex)
})` in core/store/index.ts

`once('__VUE_EXTEND_I18N__', () => {
  Vue.use(VueI18n)
})` in core/i18n/index.ts

If I uncomment this it works, @patzick : can you have a look?

**Comment by pkarw** [comment]
@mdesmet we need once() to avoid memory leaks in the ssr there must be some other solution to this failing check

**Comment by pkarw** [comment]
I belive we just must adjust the unit test: core/modules/cart/test/unit/store/mutations.spec.ts @lukeromanowicz

**Comment by lukeromanowicz** [comment]
@mdesmet I've fixed the failing test but there still were some errors coming from yours. To speed things up I've done the following changes:
- split test suite from one integration suite into 4 separate unit test suites
- fixed eslint in mixins and test
- added return values to mixin methods so they can be tested.

I hope that you don't mind that.

**Comment by lukeromanowicz** [comment]
I've also updated the Changelog and Braking Changes list because the code of this PR is not backward compatible.

**Comment by lukeromanowicz** [comment]
I approve the code but I'm not really able to fully test it. Maybe @patzick could take a look if it works as expected. Please, remember that there is also API part: https://github.com/DivanteLtd/vue-storefront-api/pull/212

File: core/compatibility/components/blocks/MyAccount/MyNewsletter.js [hunk 1]
```diff
@@ -1,35 +1,7 @@
+import { Newsletter } from '@vue-storefront/core/modules/newsletter/components/Newsletter'
+
```
Comment (filrak) [inline core/compatibility/components/blocks/MyAccount/MyNewsletter.js]: i guess we can get rid of this file and use module directly

**Comment by pkarw** [comment]
@mdesmet could You please resolve the conflicts in order to merge this feature in?

File: core/compatibility/components/blocks/MyAccount/MyNewsletter.js (hunk 1 above)
Comment (mdesmet) [reply core/compatibility/components/blocks/MyAccount/MyNewsletter.js]: OK will have a look at it tomorrow!

Comment (mdesmet) [reply core/compatibility/components/blocks/MyAccount/MyNewsletter.js]: I have removed the file and imported from the module.

**Pull request merged**